src/ReachSets/ReachTube.jl

```diff
@@ -0,0 +1,16 @@
+"""
+    ReachTube{S<:LazySet, N<:Real}
+
+Type that wraps a reach tube, which is a set together with time information.
+
+### Fields
+
+- `X`       -- set
+- `t_start` -- time interval lower bound
+- `t_end`   -- time interval upper bound
+"""
+struct ReachTube{S<:LazySet, N<:Real}
+    X::S
```

Comment:
i would have expected `X::Vector{S}` for the reach *tube*

Reply:
Hm, what is the name for `X(k)` then?

Reply:
<img width="414" alt="screen shot 2018-10-03 at 11 02 38" src="https://user-images.githubusercontent.com/12424594/46415603-ff78d500-c6fb-11e8-8ea8-d8a8467cdeb7.png">

`X(k)` could (more or less incorrectly) called `ReachSet`.

Reply:
Conceptually: if the solution of the hybrid system is one `ReachSolution`, this solution is composed of `ReachTubes`, each `ReachTube` is a vector of `ReachSets`.

Reply:
[The current draft concatenates the reach tubes](https://github.com/JuliaReach/Reachability.jl/pull/228/files#diff-36711e348dff6f0a056a775b1776623dR298) via `ReachSolution(vcat(rset...), options)`. 

I think that `vcat(rset...)` is suboptimal w.r.t. pushing one by one the computed `rset`. Doesn't `vcat(rset...)` create a temporary array?

Reply:
For me, this paragraph reads like a *reach tube* is a set of states for a time interval, so it would fit. But then you cannot differentiate between these two concepts you proposed. So I will change it.

Reply:
> The current draft concatenates the reach tubes via ReachSolution(vcat(rset...), options).
I think that vcat(rset...) is suboptimal w.r.t. pushing one by one the computed rset. Doesn't vcat(rset...) create a temporary array?

I did not write that part. But what does that have to do with this PR discussion?

Reply:
I wrote it but i wasn't very convinced with using splatting there, since i think is suboptimal and can be many sets.

If you use that `ReachTube` are vectors of sets, we can do

```julia
function hybrid_reach
    ...
    sol_hybrid_reach = ReachSolution() # empty or preallocate with a sizehint, do we know the size of its Xk in advance? i don't think so because it is related to the number of transitions

    sol_cont_reach = reach(...) # solve the continuous reach

    push!(sol_hybrid_reach.Xk, sol_cont_reach.Xk) # put the reach tube of the continuous in the set of reach tubes of the hybrid
    ....
    return  sol_hybrid_reach
```

Reply:
I'm OK with the current version (ReachSolution as a vector of ReachSets).

Reply:
Does that mean we can merge?

Reply:
Please do.